### PR TITLE
Cleanup new-IR based constant propagation

### PR DESCRIPTION
**Pull request by gramalingam**
Factor out some common logic between rewriter and constant-propagation into a utility function, and other minor cleanup.

### CODE REVIEW

File: onnxscript/ir/_convenience.py [hunk 1]
```diff
@@ -395,3 +395,44 @@ def create_value_mapping(graph: _core.Graph) -> dict[str, _core.Value]:
...
+
+
+def replace_nodes_and_values(
+    graph_or_function: _core.Graph | _core.Function,
+    insertion_point: _core.Node,
+    old_nodes: list[_core.Node],
+    new_nodes: list[_core.Node],
+    old_values: list[_core.Value],
+    new_values: list[_core.Value],
+):
```
Comment (justinchuby) [inline onnxscript/ir/_convenience.py:407]:
```suggestion
) -> None:
```

File: onnxscript/ir/_convenience.py [hunk 2]
```diff
@@ -395,3 +395,44 @@ def create_value_mapping(graph: _core.Graph) -> dict[str, _core.Value]:
                 continue
             values[value.name] = value
     return values
+
+
+def replace_nodes_and_values(
+    graph_or_function: _core.Graph | _core.Function,
```
Comment (justinchuby) [inline onnxscript/ir/_convenience.py:401]:
```suggestion
    graph_or_function: _core.Graph | _core.Function,
    /,
```

File: onnxscript/ir/_convenience.py [hunk 3]
```diff
@@ -395,3 +395,44 @@ def create_value_mapping(graph: _core.Graph) -> dict[str, _core.Value]:
                 continue
             values[value.name] = value
     return values
+
+
+def replace_nodes_and_values(
```
Comment (justinchuby) [inline onnxscript/ir/_convenience.py:400]: Helpful to have a test?

**Review by justinchuby — APPROVED** [review]

File: onnxscript/ir/_convenience.py [hunk 4]
```diff
@@ -395,3 +395,44 @@ def create_value_mapping(graph: _core.Graph) -> dict[str, _core.Value]:
...
+    graph_or_function: _core.Graph | _core.Function,
+    insertion_point: _core.Node,
+    old_nodes: list[_core.Node],
+    new_nodes: list[_core.Node],
+    old_values: list[_core.Value],
+    new_values: list[_core.Value],
+):
+    """Replaces nodes and values in the graph or function.
```
Comment (justinchuby) [inline onnxscript/ir/_convenience.py:408]: Helpful to describe the invariance this function maintains?

File: onnxscript/ir/_convenience.py [hunk 5]
```diff
@@ -395,3 +395,44 @@ def create_value_mapping(graph: _core.Graph) -> dict[str, _core.Value]:
...
     return values
+
+
+def replace_nodes_and_values(
+    graph_or_function: _core.Graph | _core.Function,
+    insertion_point: _core.Node,
+    old_nodes: list[_core.Node],
+    new_nodes: list[_core.Node],
+    old_values: list[_core.Value],
+    new_values: list[_core.Value],
```
Comment (justinchuby) [inline onnxscript/ir/_convenience.py:406]:
```suggestion
    old_nodes: Sequence[_core.Node],
    new_nodes: Sequence[_core.Node],
    old_values: Sequence[_core.Value],
    new_values: Sequence[_core.Value],
```

File: onnxscript/ir/_convenience.py (hunk 2 above)
Comment (gramalingam) [reply onnxscript/ir/_convenience.py:401]: Done (but not pushed yet). But is there a specific principle being used here, to distinguish between the two classes of parameters? Except for the name, which is admittedly not great (we could do with a better alternative to graph_or_function ... how does `container` sound? or `graph_like` ?)

Comment (justinchuby) [reply onnxscript/ir/_convenience.py:401]: I like `graph_like`!

> is there a specific principle being used here, to distinguish between the two classes of parameters

not sure what you mean? could you say more?

**Comment by justinchuby** [comment]
Is there a way to copy the metadata_props from old nodes as well?

File: onnxscript/ir/_convenience.py (hunk 2 above)
Comment (gramalingam) [reply onnxscript/ir/_convenience.py:401]: I mean, why do you prefer that graph_or_function be positional, while the others are allowed to be positional or by keyword?

**Pull request merged**